File: 30-days-of-code-challenges/day_23_bst_level_order_traversal.py
import sys

class Node:
    def __init__(self,data):
        self.right=self.left=None
        self.data = data
		
class Solution:
    def insert(self,root,data):
        if root==None:
            return Node(data)
        else:
            if data<=root.data:
                cur=self.insert(root.left,data)
                root.left=cur
            else:
                cur=self.insert(root.right,data)
                root.right=cur
        return root

	def levelOrder(self,root):
  	     #Write your code here
         if root != None:
            q = [root]
            while q:
                currentNode = q.pop()
                print currentNode.data ,
                if (currentNode.left != None):
                    q.insert(0,currentNode.left)
                if (currentNode.right != None):
                    q.insert(0,currentNode.right)    

T=int(raw_input())
myTree=Solution()
root=None
for i in range(T):
    data=int(raw_input())
    root=myTree.insert(root,data)
myTree.levelOrder(root)					

File: 30-days-of-code-challenges/day_23_bst_level_order_traversal.cpp
#include <iostream>
#include <cstddef>
#include <queue>
#include <string>
#include <cstdlib>

using namespace std;	
class Node{
    public:
        int data;
        Node *left,*right;
        Node(int d){
            data=d;
            left=right=NULL;
        }
};
class Solution{
    public:
  		Node* insert(Node* root, int data){
            if(root==NULL){
                return new Node(data);
            }
            else{
                Node* cur;
                if(data<=root->data){
                    cur=insert(root->left,data);
                    root->left=cur;
                }
                else{
                   cur=insert(root->right,data);
                   root->right=cur;
                 }           
           return root;
           }
        }
		
		void levelOrder(Node * root){
			//Write your code here
			queue<Node*> q;
			if(root) {
				q.push(root);
				while(!q.empty()) {
					Node *currentNode = q.front();
					q.pop();
					cout << currentNode->data << " ";
					if(currentNode->left) {
						q.push(currentNode->left);
					}
					if(currentNode->right) {
						q.push(currentNode->right);
					}
				}
			}  
		}
};//End of Solution

int main(){
    Solution myTree;
    Node* root=NULL;
    int T,data;
    cin>>T;
    while(T-->0){
        cin>>data;
        root= myTree.insert(root,data);
    }
    myTree.levelOrder(root);
    return 0;
}